design/workitems.go

```diff
@@ -150,4 +150,23 @@ var _ = a.Resource("workitem", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("reorder", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/reorder/:id?before=:before"),
+		)
+		a.Description("reorder the work item with the given id.")
+		a.Params(func() {
+			a.Param("id", d.String, "ids")
+			a.Param("before", d.String, "before id")
+		})
+		a.Payload(workItemSingle)
```

Comment:
@kwk There is a payload for reordering because the **version** of the to-be-reordered workitem changes. The **version** should change because one of the fields' value i.e. **SystemOrder** changes.